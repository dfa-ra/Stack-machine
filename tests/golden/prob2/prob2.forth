_data_
0 VAR n
0 VAR sum_pow2
0 VAR pow2_sum

0x80 VAR input_addr
0x84 VAR output_addr

_func_
: PRINT_NUM
    output_addr a! !
;
: READ_NUM
    input_addr a! @
;
: SUM
    LOOP
        1 !j n <= 1
        WHILE
            sum_pow2 j + !sum_pow2
        REPEAT
;
: SUM_POW2
    LOOP
        1 !i n <= 1
        WHILE
            i i * !tmp
            pow2_sum tmp + !pow2_sum
        REPEAT
;


_text_
READ_NUM !n

SUM
sum_pow2 sum_pow2 * !sum_pow2
SUM_POW2
pow2_sum sum_pow2 - !n
@n PRINT_NUM

HALT
