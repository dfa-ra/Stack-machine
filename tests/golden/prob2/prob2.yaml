name: sort
limit: 5000
memory_size: 500
input_streams:
  0x80: [100]
  0x84: []
reports:
  - name: binary
    slice: mem
    view: |
      {instruction}
  - name: Step-by-step
    slice: all
    view: |
      tick: {cpu:tick} || microcode - {microcode} || A: {cpu:A:hex} || B: {cpu:B:hex} 
      data stack: {stack:data}
  - name: Result
    slice: last
    view: |
      numio[0x80]: {io:in:dec}
      numio[0x84]: {io:out:decbe}
      tick_count = {cpu:tick}
      {condition:[io:out:decbe]=[25164150]}